Automattic/wp-calypso
# Themes: Remove redundant actions and reducers

No visual changes. This PR removes redundant code.

It is not necessary to have both THEMES_RECEIVE and THEMES_REQUEST_SUCCESS
action types, since they both signify an array of themes being returned
from a server.

Remove THEMES_RECEIVE and allow the receiveThemes() action to accept
a query, allowing it to take the place of receiveThemesQuery().

**To Test**
Go to http://calypso.localhost:3000/design
* Check that theme searches and queries work exactly as before, for jetpack and non-jetpack sites
* Check that activating a theme works as before
* Check that the _CURRENT THEME_ bar at the top of the theme showcase works as before
